packages/openapi-generator/src/parser/schema.ts

```diff
@@ -221,11 +223,46 @@ function parseXOfSchema(
   xOf: 'oneOf' | 'allOf' | 'anyOf',
   options: ParserOptions
 ): any {
+  const normalizedSchema = normalizeSchema(schema, xOf);
+
   return {
-    [xOf]: (schema[xOf] || []).map(entry => parseSchema(entry, refs, options))
+    [xOf]: (normalizedSchema[xOf] || []).map(entry =>
+      parseSchema(
+        {
+          ...entry,
+          required: [
+            // Add required properties from the entry.
+            ...('required' in entry && entry.required ? entry.required : []),
+            // Add required properties from the top level schema (xOf).
+            ...(normalizedSchema.required || [])
+          ]
+        },
+        refs,
+        options
+      )
+    )
   };
 }
 
+function normalizeSchema(
+  schema: OpenAPIV3.NonArraySchemaObject,
+  xOf: 'oneOf' | 'allOf' | 'anyOf'
+): OpenAPIV3.NonArraySchemaObject {
+  if (
+    schema.properties ||
+    (schema.additionalProperties &&
+      typeof schema.additionalProperties === 'object')
```

Comment:
if the answer to the question above is yes, you could consider extracting a function to check whether the given schema is an object schema and reusing it for the object check above as well.

Reply:
So, if the object has properties and `schema.additionalProperties = true` then it will add it. If its:

```
type: object
allOf:
  - ...
additionalProperties: true
```

then I think not. I probably can just check for `schema.additionalProperties` instead of ```
schema.additionalProperties && typeof schema.additionalProperties === 'object'
```

and that should already fix it. Honestly I am forgetting now why I checked for object type. Let me try.

Reply:
Tested this after removing `&& typeof schema.additionalProperties === 'object'` check. And it works.